src/snowflake/cli/_plugins/stage/manager.py

```diff
@@ -320,9 +327,14 @@ def execute(
     ):
         stage_path_parts = self._stage_path_part_factory(stage_path)
         all_files_list = self._get_files_list_from_stage(stage_path_parts)
+        all_files_with_stage_name_prefix = [
+            stage_path_parts.add_stage_prefix(file) for file in all_files_list
+        ]
```

Comment:
With small stages is not a problem but with git repo stage, where there are tens or hundreds files, it is huge waste. I would avoid it.